gcpy/file_regrid.py

```diff
@@ -448,157 +551,457 @@ def regrid_ll_to_cssg(
-        non_fields_ds = ds_in[non_fields]
-        ds_in = ds_in.drop(non_fields)
+        dset = dset.drop(["lat_bnds", "lon_bnds"])
+        non_fields = dset[non_fields]
+        dset = dset.drop(non_fields)
 
-        # Create the regridder
+        # Create the regridder and regrid the data
         regridder = make_regridder_L2L(
         ll_res_in,
             ll_res_out,
             reuse_weights=True,
             in_extent=in_extent,
-            out_extent=out_extent
+            out_extent=out_extent,
+            weightsdir=weightsdir
+        )
+        dset = regridder(
+            dset,
+            keep_attrs=True
         )
-        ds_out = regridder(ds_in, keep_attrs=True, verbose=False)
 
         # Add the non-regriddable fields back
-        ds_out = ds_out.merge(non_fields_ds)
+        dset = dset.merge(non_fields)
 
         # Change order of dimensions
-        ds_out = ds_out.transpose(
-            'time', 'lev', 'ilev', 'lat', 'lon', ...
+        dset = dset.transpose(
+            "time", "lev", "ilev", "lat", "lon", ...
+        )
+
+        # Set the lev:positive attribute accordingly
+        dset = flip_lev_coord_if_necessary(
+            dset,
+            dim_format_in="classic",
+            dim_format_out="classic"
+        )
+
+        # Save lat/lon coordinate metadata
+        dset = save_ll_metadata(
+            dset,
+            verbose=verbose
         )
 
-    return ds_out
+    return dset
+
+
+def flip_lev_coord_if_necessary(
+        dset,
+        dim_format_in,
+        dim_format_out
+):
+    """
+    Determines whether it is necessary to flip the "lev" and "ilev"
+    coords of an xarray.Dataset in the vertical depending on the
+    values ofdim_format_in and dim_format_out.  Also sets the
+    attributes "lev:positive" and "ilev:positive" accordingly.
+
+    Args:
+    -----
+    dset : xarray.Dataset
+        The input dataset.
+    dim_format_in : str
+        Input grid format ("classic", "checkpoint", "diagnostic").
+    dim_format_out : str
+        Output grid format ("classic", "checkpoint", "diagnostic").
+
+    Args:
+    -----
+    dset : xarray.Dataset
+        The modified dataset.
+
+    Remarks:
+    --------
+    (1) classic    : lev is in ascending order  (lev:positive="up"  )
+    (2) diagnostic : lev is in ascending order* (lev:positive="up"  )
+    (3) checkpoint : lev is in descending order (lev:positive="down")
+
+    *Except for the Emissions collection, which has lev arranged
+     in descending order.
+    """
+    verify_variable_type(dset, xr.Dataset)
+    verify_variable_type(dim_format_in, str)
+    verify_variable_type(dim_format_out, str)
+
+    # ==================================================================
+    # checkpoint/diagnostic to classic: lev, ilev in ascending order
+    # ==================================================================
+    if dim_format_in != "classic" and dim_format_out == "classic":
```

Comment:
Shouldn't checkpoint to classic and diagnostic to classic be handled differently?

Reply:
@lizziel: I think this will fix it:
```python
    # ==================================================================
    # Case 1 :checkpoint/diagnostic to classic
    # lev, ilev need to be in ascending order
    # ==================================================================
    if dim_format_in != "classic" and dim_format_out == "classic":

        # Flip lev and set to eta values at midpoints (if necessary)
        if "ilev" in dset.coords:
            if is_gchp_lev_positive_down(dset):
                dset = dset.reindex(ilev=dset.ilev[::-1])
            if any(var > 1.0 for var in dset.ilev):
                coord = get_ilev_coord(
                    n_lev=dset.dims["ilev"],
                    top_down=False
                )
                dset = dset.assign_coords({"ilev": coord})
            dset.ilev.attrs["positive"] = "up"

        # Flip lev and set to eta values at midpoints (if necessary)
        if "lev" in dset.coords:
            if is_gchp_lev_positive_down(dset):
                dset = dset.reindex(lev=dset.lev[::-1])
            if any(var > 1.0 for var in dset.lev):
                coord = get_lev_coord(
                    n_lev=dset.dims["lev"],
                    top_down=False
                )
                dset = dset.assign_coords({"lev": coord})
            dset.lev.attrs["positive"] = "up"

        return dset
```

Reply:
@lizziel: I tried this code:
```python
        # Flip lev and set to eta values at midpoints (if necessary)
        if "lev" in dset.coords:
            if is_gchp_lev_positive_down(dset):
                dset = dset.reindex(lev=dset.lev[::-1])
            if any(var > 1.0 for var in dset.lev):
                coord = get_lev_coord(
                    n_lev=dset.dims["lev"],
                    top_down=False
                )
                dset = dset.assign_coords({"lev": coord})
            dset.lev.attrs["positive"] = "up"
```
and for checkpoint -> classic it works OK.  But when I do diagnostic -> classic I get this:
<img width="693" alt="Screenshot 2023-10-11 at 2 11 51 PM" src="https://github.com/geoschem/gcpy/assets/5880296/8e8295f9-ceed-4aa9-98b3-385aae103ef8">


Do you know if there is a flipping in the vertical in plotting?  I wouldn't think so but you never know.

Reply:
Maybe we can just say that diagnostic regridding is unreliable and kick the can down the road

Reply:
> Maybe we can just say that diagnostic regridding is unreliable and kick the can down the road
Agreed!